src/CssInliner.php

```diff
@@ -1012,39 +1011,40 @@ private function removeUnmatchablePseudoComponents(string $selector): string
 
         // The regex allows nested brackets via `(?2)`.
         // A space is temporarily prepended because the callback can't determine if the match was at the very start.
-        $selectorWithoutNots = \ltrim((new Preg())->throwExceptions($this->debug)->replaceCallback(
-            '/([\\s>+~]?+):not(\\([^()]*+(?:(?2)[^()]*+)*+\\))/i',
-            /** @param array<array-key, string> $matches */
-            function (array $matches): string {
-                return $this->replaceUnmatchableNotComponent($matches);
-            },
-            ' ' . $selector
-        ));
+        $pattern = '/([\\s>+~]?+):not(\\([^()]*+(?:(?2)[^()]*+)*+\\))/i';
+        /** @param array<array-key, string> $matches */
+        $callback = function (array $matches): string {
+            return $this->replaceUnmatchableNotComponent($matches);
+        };
+        $selectorWithoutNots = (new Preg())->throwExceptions($this->debug)
+            ->replaceCallback($pattern, $callback, ' ' . $selector);
+        $trimmedSelectorWithoutNots = \ltrim($selectorWithoutNots);
 
         $selectorWithoutUnmatchablePseudoComponents = $this->removeSelectorComponents(
             ':(?!' . self::PSEUDO_CLASS_MATCHER . '):?+[\\w\\-]++(?:\\([^\\)]*+\\))?+',
-            $selectorWithoutNots
+            $trimmedSelectorWithoutNots
         );
 
-        if (
-            $preg->match(
-                '/:(?:' . self::OF_TYPE_PSEUDO_CLASS_MATCHER . ')/i',
-                $selectorWithoutUnmatchablePseudoComponents
-            )
-            === 0
-        ) {
+        if ($preg->match(
+            '/:(?:' . self::OF_TYPE_PSEUDO_CLASS_MATCHER . ')/i',
+            $selectorWithoutUnmatchablePseudoComponents
+        ) === 0) {
             return $selectorWithoutUnmatchablePseudoComponents;
         }
+
+        $selectorParts = $preg->split(
+            '/(' . self::COMBINATOR_MATCHER . ')/',
+            $selectorWithoutUnmatchablePseudoComponents,
+            -1,
+            PREG_SPLIT_DELIM_CAPTURE | PREG_SPLIT_NO_EMPTY
+        );
+        /** @var list<string> $selectorParts */
```

Comment:
Is this needed?  `Preg::split()` is declared as returning `array<int, string>` (and could be tightened to `list<string>`), though `array_map()` does not require a sequential array.